### DISCUSSION TITLE
Fixed set of peer without any discovery

### DISCUSSION THREAD

**Discussion by sandeshbhatjr** [anacrolix/torrent#877] [General]
I am looking at using the torrent protocol for moving around files between known peers. I do not want it to use trackers, Peer Exchange, DHT or LSD. I have a fixed list of peers with their addresses (IP, port) that I synchronise using my own protocol. Can I  use the AddPeer() method for torrent in this library to achieve this?

**Comment by anacrolix** [comment]
Yes.

**Reply by sandeshbhatjr** [reply]
@anacrolix In the case where AddPeers adds a peer that is not yet online, and takes a while to come online, does the torrent client check if the peer is available at regular intervals and connect to it when available?

**Reply by anacrolix** [reply]
No, once it's attempted it gets forgotten, whether it succeeded or not.

**Reply by sandeshbhatjr** [reply]
Is there any recommended way to know when the outgoing connection to peer fails and handle adding it again? For now, I just add the same peers at fixed intervals repeatedly and this works. But it seems a bit hacky. Ideally, I would like to react to events of connection failures.

**Reply by anacrolix** [reply]
Not currently. I add in a loop too, in several apps and some of the tests.

If you can share more about your use case I could figure out if there's an appropriate feature to add.

I have considered smarter management of peers, like adding good peers back to the pool if they disconnect, but it quickly becomes non trivial.

**Reply by sandeshbhatjr** [reply]
I use your library to transfer large files across many peers with data flow between peers restricted by a customisable topology (for example, in a daisychained fashion but not necessarily). I don't need the peer discovery part of the bit torrent protocol, as the custom topology provider provides the list of peers to connect to for each peer. Now, it is possible that some of these peers go down from time to time, hence the need to reconnect once they are back online.

I use bit torrent because it's highly tuned to faciliate chunking and transfer of these chunks via pipelining and all that, instead of hand rolling my own UDP transfer mechanism.
